backend/src/main/java/de/bund/digitalservice/ris/norms/utils/NodeParser.java

```diff
@@ -36,9 +37,15 @@ public static Optional<String> getValueFromExpression(String xPathExpression, No
         return Optional.empty();
       }
 
+      // Clone the node if the xpath is relative to the node. This drastically improves the
+      // performance.
+      if (xPathExpression.startsWith("./")) {
+        sourceNode = sourceNode.cloneNode(true);
+      }
+
       // should be invoked on every method call since: An XPath object is not thread-safe and not
       // reentrant.
-      final XPath xPath = XPathFactory.newInstance().newXPath();
+      final XPath xPath = xPathFactory.newXPath();
```

Comment:
Could you also add the same `xPathFactory` instance to the other methods? I guess only the `XPath` object in not thread-safe and not the factory instance correct?

Reply:
Oh yes I'll definitely do that. And yes only the XPath itself is not thread safe (and can be created in a very small fraction of the time it takes to create a factory).